### PR TITLE
feat(prover): use httptest.Server to simplify the prover web server test code

### PR DESCRIPTION
**Pull request by alexshliu**

### CODE REVIEW

**Review by RogerLamTd — APPROVED** [review]

File: prover/server/server.go [hunk 1]
```diff
@@ -73,11 +73,6 @@ func (srv *ProverServer) Shutdown(ctx context.Context) error {
 	return srv.echo.Shutdown(ctx)
 }
 
-// ServeHTTP implements the `http.Handler` interface which serves HTTP requests.
-func (srv *ProverServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```
[the remark this answers is not in the document]
Comment (alexshliu) [reply prover/server/server.go:77]: This is just used by previous test.

[the remark this answers is not in the document]
Comment (alexshliu) [reply prover/server/server.go:77]: @cyberhorsey pls have a look.

**Comment by cyberhorsey** [comment]
can you resolve the conflicts @alexshliu

**Comment by alexshliu** [comment]
> can you resolve the conflicts @alexshliu

fixed

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**